main.go

```diff
@@ -376,6 +376,10 @@ func getFeatureLabels(source source.FeatureSource) (labels Labels, err error) {
 
 		// Validate label value
 		value := fmt.Sprintf("%v", v)
+		if len(value) > 63 {
```

Comment:
I would declare constant like in `pkg/apis/core/validation/validation.go`, so it will be understandable where this 63 value came from.
```go
const qualifiedNameMaxLength int = 63
```
Otherwise, lgtm.

Reply:
Hmm, actually why not use validation from apimachinery directly.

I'll get back to this...

Reply:
I changed this patch to use apimachinery directly for all validation